numpy/core/function_base.py

```diff
@@ -142,7 +142,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     y += start
 
     if endpoint and num > 1:
-        y[-1] = stop
+        y[-1,...] = stop
```

Comment:
Why is this needed? Seems the problem is in the callers.

Reply:
If `stop` is a scalar, this works fine without `...`. However, linspace also accepts `Arrayish` inputs for `start` and `stop`. See https://github.com/numpy/numpy/issues/3504.